### PR TITLE
apport-retrace: exit without 0 on error

### PR DESCRIPTION
**Pull request by bdrung**
Exit `apport-retrace` with return code 2 in case the report is malformed and with return code 3 on known errors. Crashes of `apport-retrace` will exit with return code 1.

### CODE REVIEW

File: bin/apport-retrace [hunk 1]
```diff
@@ -349,7 +349,7 @@ def load_report(
         except AssertionError as error:
             if "apport format data" in str(error):
                 apport.error("Broken report: %s", str(error))
-                sys.exit(0)
+                sys.exit(2)
```
Comment (schopin-pro) [inline bin/apport-retrace:352]: issue: it's a shame to have main() return an int but still have a hard exit() call in `load_report()`. How about throwing an exception instead, in proper Python fashion?

File: bin/apport-retrace [hunk 2]
```diff
@@ -391,15 +391,15 @@ Thank you for your understanding, and sorry for the inconvenience!
 
 
 # pylint: disable-next=missing-function-docstring
-def main():
+def main(argv):
```
Comment (schopin-pro) [inline bin/apport-retrace:394]: praise: Nice, makes it easier to test!

**Review by schopin-pro — CHANGES_REQUESTED** [review]
Very  nice, except for the load_report() bit where we're missing the opportunity to remove a raw exit() call.

File: bin/apport-retrace (hunk 1 above)
Comment (bdrung) [reply bin/apport-retrace:352]: Okay. Introduced `_LoadReportException`.

**Review by schopin-pro — APPROVED** [review]

**Pull request merged**